src/main/java/de/tum/in/www1/artemis/repository/ComplaintRepository.java

```diff
@@ -90,9 +95,12 @@ public interface ComplaintRepository extends JpaRepository<Complaint, Long> {
      * @return the number of unaccepted complaints
      */
     @Query("""
-            SELECT COUNT(c) FROM Complaint c
-            WHERE c.complaintType = 'COMPLAINT' AND c.student.id = :#{#studentId}
-            AND c.result.participation.exercise.course.id = :#{#courseId} AND (c.accepted = false OR c.accepted is null)
+            SELECT COUNT(c)
+            FROM Complaint c
+            WHERE c.complaintType = de.tum.in.www1.artemis.domain.enumeration.ComplaintType.COMPLAINT
+                AND c.student.id = :studentId
+                AND c.result.participation.exercise.course.id = :courseId
+                AND (c.accepted IS FALSE OR c.accepted IS NULL)
             """)
     long countUnacceptedComplaintsByComplaintTypeStudentIdAndCourseId(@Param("studentId") Long studentId, @Param("courseId") Long courseId);
```

Comment:
same for the methods below. With `byComplaintType` I'd expect to be able to pass the wanted type as a parameter.
```suggestion
    long countUnacceptedComplaintsByStudentIdAndCourseId(@Param("studentId") Long studentId, @Param("courseId") Long courseId);
```